### PR TITLE
Feature/update signup process

### LINKED ISSUE #148: [Feature] Add more fields to `Auth/Signup`
### Description

As of now we are saving only `username`, `email`, and `password` to the Supabase table. Hence we need to save more data to reflect them back in the `User Profile`

### Tasks
- [ ] Update the `SignUpSchema` in `utils/schema.ts` with the remaining fields
   - FirstName
   - LastName
   - Gender
   - Phone 
   - Address
   - Account Creation Date (Use `new Date();`)

### Screenshots

_The following is the supabase table you'd need to work upon. The column names can be found in the Screenshot below._

image

### Checklist

- [X] I have checked the existing [issues](https://github.com/pooranjoyb/popShop/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/pooranjoyb/popShop/blob/main/CONTRIBUTING.md)
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by me-shweta**
In **utils/schema.ts**, I updated the SignUpSchema to include additional fields like FirstName, LastName, Gender, Phone, Address, and Account Creation Date. In **auth.ts**, I integrated the updated schema to validate user signup data in the authentication component.

Fixes #148 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# Checklist:
give a X on it which is applicable

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @me-shweta! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by pooranjoyb** [comment]
There are build errors. Fix them and commit. Run `npm run build` to detect the errors in the code. 

image

@me-shweta

File: src/pages/Auth/Auth.tsx
Comment (pooranjoyb) [inline src/pages/Auth/Auth.tsx:1]: Hii @me-shweta I don't see much changes that you did here. 
Also `interface USER` is not updated.

File: src/utils/schema.ts
Comment (pooranjoyb) [inline src/utils/schema.ts:1]: `schema` definition is not proper. :/

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

**Comment by me-shweta** [comment]
@pooranjoyb Tell me, is there any problem still? And I'm just reminding, if everything is fine, don't forget to add labels to this pull request.

File: package.json
Comment (pooranjoyb) [inline package.json:1]: I didn't understand why did you added `jest` for testing.

File: src/components/Button.tsx
Comment (pooranjoyb) [inline src/components/Button.tsx:1]: I guess refactoring `Button.jsx` wasn't a part of the issue.

File: src/utils/schema.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { z } from "zod";
 
+// SignUpSchema
 const SignUpSchema = z.object({
   username: z
     .string()
-    .min(3, "Username must be at least 3 characters")
```
Comment (pooranjoyb) [inline src/utils/schema.ts:6]: Doesn't need a `message` key.

File: vite.config.ts
Comment (pooranjoyb) [inline vite.config.ts:1]: Revert this. Unnecessary changes.

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

**Comment by pooranjoyb** [comment]
Please read `CONTRIBUTING GUIDELINES` seems like you didn't run `npm run watch-css` for tailwind. This eventually broke the UI :/ @me-shweta 

image

**Comment by me-shweta** [comment]
@pooranjoyb I have changed the code of the files, let me know if the problem still persists. :_)

**Comment by pooranjoyb** [comment]
Seems like there are lotta issues here, even the Ui is not fixed. So, looking at your efforts to fix it, I'm merging it to `test` branch :/ But sorry to say that can't merge into `dev`

You'll still get the `gssoc` label but can't assign `level-2` to this :/ @me-shweta

**Comment by me-shweta** [comment]
@pooranjoyb Sorry, I couldn't manage to do it in time. I think I should stay away from coding issues. By the way, thank you very much for your support. :/

**Comment by pooranjoyb** [comment]
> @pooranjoyb Sorry, I couldn't manage to do it in time. I think I should stay away from coding issues. By the way, thank you very much for your support. :/
> By the way, these PR points will be considered, right? Or do I need to be assigned as the PR's assignee as well?

Yes it will be considered as a valid contribution, don't worry. And yes, I'd say try with easier UI issues. Don't just give up with bugs/problems easily. Try to understand how things actually work, use chatgpt to understand codebase. This will really help you understand better. @me-shweta

**Pull request merged**